In [184]:
import pandas as pd
import numpy as np
import unidecode

In [185]:
# Lee el archivo sin ninguna fila de encabezado
df = pd.read_csv("combustibles.csv", header=None)

# Define manualmente los nombres de las columnas basándote en tu conocimiento del archivo
# Si necesitas combinar varias filas para los nombres de las columnas, tendrás que hacerlo manualmente
column_names = ['N°', 
                'Fecha de Verificacion', 
                'Número de PERMISO otorgado por la CRE', 
                'Número de Estacion', 
                'Razón Social', 
                'ESTADO', 
                'MUNICIPIO', 
                'TOTAL DE INSTUMENTOS (MANGUERAS) VERIFICADOS', 
                'TOTAL DE INSTUMENTOS (MANGUERAS) INMOVILIZADOS', 
                'CAUSAS DE INMOVILIZACION Acreditación documental (1)',
                'CAUSAS DE INMOVILIZACION Calidad del Combustible (2)',
                'CAUSAS DE INMOVILIZACION Cualidades Metrólogicas (3) 3.1',
                'CAUSAS DE INMOVILIZACION Cualidades Metrólogicas (3) FALTANTE MAXIMO PROMEDIO POR LITRO (EN ml)',
                'CAUSAS DE INMOVILIZACION Cualidades Metrólogicas (3) EXCEDENTE MAXIMO PROMEDIO POR LITRO (EN ml)',
                'CAUSAS DE INMOVILIZACION Cualidades Metrólogicas (3) 3.2',
                'CAUSAS DE INMOVILIZACION Causas Electrónicas (4) 4.1',
                'CAUSAS DE INMOVILIZACION Causas Electrónicas (4) 4.2',
                'CAUSAS DE INMOVILIZACION Causas Electrónicas (4) 4.3',
                'CAUSAS DE INMOVILIZACION Causas Electrónicas (4) 4.4',
                'CAUSAS DE INMOVILIZACION Causas Electrónicas (4) 4.5',
                'CAUSAS DE INMOVILIZACION Causas Electrónicas (4) 4.6',
                'CAUSAS DE INMOVILIZACION Causas Electrónicas (4) 4.7',
                'CAUSAS DE INMOVILIZACION Causas Electrónicas (4) 4.8',
                'CAUSAS DE INMOVILIZACION Causas Electrónicas (4) 4.9',
                'CAUSAS DE INMOVILIZACION Causas Electrónicas (4) 4.10',
                'CAUSAS DE INMOVILIZACION Fallas Hidráulicas (5) 5.1',
                'CAUSAS DE INMOVILIZACION Fallas Hidráulicas (5) 5.2',
                'CAUSAS DE INMOVILIZACION Falta o deficiencias en servicio de calibración (6) 6.1',
                'CAUSAS DE INMOVILIZACION Falta o deficiencias en servicio de calibración (6) 6.2',
                'CAUSAS DE INMOVILIZACION Falta o deficiencias en servicio de calibración (6) 6.3',
                'CAUSAS DE INMOVILIZACION Falta o deficiencias en servicio de calibración (6) DISTINTIVO EMPRESARIAL (6.4?)',
                'CAUSAS DE INMOVILIZACION Falta o deficiencias en servicio de calibración (6) DICTAMEN (6.5?)',
                'CAUSAS DE INMOVILIZACION Falta o deficiencias en servicio de calibración (6) ( CAMBIA A 6.6?)',
                'CAUSAS DE INMOVILIZACION Fuera de servicio',
                'CAUSAS DE INMOVILIZACION Precio',
                'CAUSAS DE INMOVILIZACION Seguridad 9.1',
                'CAUSAS DE INMOVILIZACION Seguridad 9.2',
                'Dirección',
                'Colonia',
                'Código postal',
                ''
               ]
# Ahora puedes eliminar las filas que ya no necesitas, que contenían los encabezados originales
df = df.drop(index=[0, 1, 2, 3])  # Ajusta los índices de las filas según sea necesario

df.columns = column_names
df.fillna('', inplace=True)

# Restablecer el índice si las filas se eliminaron
df.reset_index(drop=True, inplace=True)

df.tail(5)

N° Fecha de Verificacion  \
4116            4117               6/25/23   
4117            4118               6/25/23   
4118                                         
4119  Hoja3: Tabla 1                         
4120                                         

     Número de PERMISO otorgado por la CRE Número de Estacion  \
4116                   PL/3551/EXP/ES/2015                955   
4117                    PL/497/EXP/ES/2015               9603   
4118                                                            
4119                                                            
4120                                                            

                      Razón Social               ESTADO      MUNICIPIO  \
4116  JAIME IGNACIO VALDES GARDUÑO               MEXICO  TEMASCALCINGO   
4117  GRUPO OCTANO, S. A. DE C. V.  MICHOACÁN DE OCAMPO    ZINAPÉCUARO   
4118                                                                     
4119                                                                     
4120                                                                     

     TOTAL DE INSTUMENTOS (MANGUERAS) VERIFICADOS  \
4116                                           14   
4117                                           20   
4118                                       64,098   
4119                                                
4120                                                

     TOTAL DE INSTUMENTOS (MANGUERAS) INMOVILIZADOS  \
4116                                              0   
4117                                              0   
4118                                          1,897   
4119                                                  
4120                                                  

     CAUSAS DE INMOVILIZACION Acreditación documental (1)  ...  \
4116                                                  0    ...   
4117                                                  0    ...   
4118                                                  6    ...   
4119                                                       ...   
4120                                                       ...   

     CAUSAS DE INMOVILIZACION Falta o deficiencias en servicio de calibración (6) DICTAMEN (6.5?)  \
4116                                                  0                                             
4117                                                  0                                             
4118                                                 16                                             
4119                                                                                                
4120                                                                                                

     CAUSAS DE INMOVILIZACION Falta o deficiencias en servicio de calibración (6) ( CAMBIA A 6.6?)  \
4116                                                  0                                              
4117                                                  0                                              
4118                                                  0                                              
4119                                                                                                 
4120                                                                                                 

     CAUSAS DE INMOVILIZACION Fuera de servicio  \
4116                                          0   
4117                                          0   
4118                                         55   
4119                                              
4120                                              

     CAUSAS DE INMOVILIZACION Precio CAUSAS DE INMOVILIZACION Seguridad 9.1  \
4116                               0                                      0   
4117                               0                                      0   
4118                              12                                     11   
4119           

# Question 1
## What percentage of stations passed all the fuel quality and metrological tests?
## ¿Qué porcentaje de estaciones pasaron todas las pruebas de calidad del combustible y metrológicas?

In [186]:
# Filter the relevant columns for the tests
test_columns = [col for col in df.columns if 'CAUSAS DE INMOVILIZACION Calidad del Combustible' in col or 'CAUSAS DE INMOVILIZACION Cualidades Metrólogicas' in col]
df_question1 = df.copy()
df_question1 = df_question1[:-3]
nan_counts = {}
for col in test_columns:
    df_question1[col] = pd.to_numeric(df_question1[col], errors='coerce')  # Coerce errors will set problematic entries to NaN
    nan_counts[col] = df_question1[col].isna().sum()

print(nan_counts)

{'CAUSAS DE INMOVILIZACION Calidad del Combustible (2)': 0, 'CAUSAS DE INMOVILIZACION Cualidades Metrólogicas (3) 3.1': 0, 'CAUSAS DE INMOVILIZACION Cualidades Metrólogicas (3) FALTANTE MAXIMO PROMEDIO POR LITRO (EN ml)': 0, 'CAUSAS DE INMOVILIZACION Cualidades Metrólogicas (3) EXCEDENTE MAXIMO PROMEDIO POR LITRO (EN ml)': 0, 'CAUSAS DE INMOVILIZACION Cualidades Metrólogicas (3) 3.2': 0}


In [187]:
# Sum the values across these columns for each row
df_question1['Total Tests'] = df_question1[test_columns].sum(axis=1)

In [188]:
# Gas stations that did not meet standards
df_filtered = df_question1[df_question1['Total Tests'] != 0]
gas_stations_test_failed = df_filtered[['Razón Social', 'Total Tests']]
gas_stations_test_failed

Razón Social  Total Tests
9           SÚPER SERVICIO GALAXIA. S. A. DE C. V.        45.89
18                  ARAS LOS REYES. S. A. DE C. V.        13.81
75          SÚPER SERVICIO RIOLUNA. S. A. DE C. V.         8.27
165   MULTI SERVICIO CHAPA DE MOTA. S. A. DE C. V.         9.86
194                         CARITINA VARGAS FLORES        18.34
...                                            ...          ...
3729           AUTOSERVICIO AURORA, S. A. DE C. V.         6.59
3820            COMBUSTIBLES MURPA, S. A. DE C. V.        40.75
3827                      PETROMAX, S. A. DE C. V.         9.55
4008        GASOLINERA LA AVENTURA, S. A. DE C. V.         7.23
4054                      PETROMAX, S. A. DE C. V.         7.31

[104 rows x 2 columns]

In [189]:
# Count how many stations passed all the tests
stations_passed = (df_question1['Total Tests'] == 0).sum()

In [190]:
# Calculate the total number of stations
total_stations = len(df_question1)

# Calculate the percentage
percentage_passed = (stations_passed / total_stations) * 100

print(f"The percentage of stations that passed all the tests is: {percentage_passed:.2f}%")

The percentage of stations that passed all the tests is: 97.47%


# Question 2
## ¿Qué estados tuvieron un mayor porcentage de gasolineras que tuvieron un faltante máximo promedio por litros?
## Error máximo tolerado
(Despacho inexacto)
La diferencia de combustible detectada entre lo indicado por el dispensario y lo entregado al consumidor.

De acuerdo a lo establecido por la Norma, para 20 litros despachados, la tolerancia será de 100 ml. y para 10 litros despachados será de 60 ml.

In [191]:
df_question_2 = df.copy()
df_question_2 = df_question_2[:-3]

In [192]:
test_columns_2 = [
    "ESTADO", 
    'CAUSAS DE INMOVILIZACION Cualidades Metrólogicas (3) 3.1',
    'CAUSAS DE INMOVILIZACION Cualidades Metrólogicas (3) FALTANTE MAXIMO PROMEDIO POR LITRO (EN ml)'
]

nan_counts_2 = {}

for col in test_columns_2:
    if col == "ESTADO":
        df_question_2['ESTADO'] = df_question_2['ESTADO'].apply(lambda x: unidecode.unidecode(x).upper())
    else:
        df_question_2[col] = pd.to_numeric(df_question_2[col], errors='coerce')  # Coerce errors will set problematic entries to NaN
        nan_counts_2[col] = df_question_2[col].isna().sum()

print(nan_counts_2)    

{'CAUSAS DE INMOVILIZACION Cualidades Metrólogicas (3) 3.1': 0, 'CAUSAS DE INMOVILIZACION Cualidades Metrólogicas (3) FALTANTE MAXIMO PROMEDIO POR LITRO (EN ml)': 0}


In [193]:
df_question_2.groupby("ESTADO").size()

ESTADO
                                     1
AGUASCALIENTES                      82
BAJA CALIFORNIA                    146
BAJA CALIFORNIA SUR                 37
CAMPECHE                            43
CHIAPAS                            126
CHIHUAHUA                          187
CIUDAD DE MEXICO                   103
COAHUILA DE ZARAGOZA                45
COLIMA                              43
DURANGO                            105
GUANAJUATO                         220
GUERRERO                            38
HIDALGO                            103
JALISCO                            347
MEXICO                             332
MICHOACAN DE OCAMPO                185
MORELOS                             73
NAYARIT                             61
NUEVO LEON                         204
OAXACA                             125
PUEBLA                             101
QUERETARO                          169
QUINTANA ROO                        39
SAN LUIS POTOSI                    120
SINALOA           

In [194]:
# Calculate the total number of gas stations for each state and add to a column gas station 
# Calculate the number of gas stations who did not passed the test and add on another column
# Calculate the percentage of gas stations that did not pass the test.

In [195]:
# Filtrar las gasolineras que no pasaron las pruebas por estado
df_no_pasaron = df_question_2[df_question_2["CAUSAS DE INMOVILIZACION Cualidades Metrólogicas (3) 3.1"] == 1]
conteo_no_pasaron_por_estado = df_no_pasaron.groupby("ESTADO").size()
conteo_no_pasaron_por_estado.head(10)

ESTADO
AGUASCALIENTES       2
BAJA CALIFORNIA      3
CHIAPAS              1
CHIHUAHUA            8
CIUDAD DE MEXICO     2
DURANGO              5
GUANAJUATO          11
GUERRERO             1
JALISCO              3
MEXICO               5
dtype: int64

In [196]:
# Obtener el conteo total por estado
conteo_total_por_estado = df_question_2.groupby("ESTADO").size()
conteo_total_por_estado.head(10)

ESTADO
                          1
AGUASCALIENTES           82
BAJA CALIFORNIA         146
BAJA CALIFORNIA SUR      37
CAMPECHE                 43
CHIAPAS                 126
CHIHUAHUA               187
CIUDAD DE MEXICO        103
COAHUILA DE ZARAGOZA     45
COLIMA                   43
dtype: int64

In [197]:
# Calcular el porcentaje de gasolineras que no pasó la prueba
porcentaje_no_pasaron = (conteo_no_pasaron_por_estado / conteo_total_por_estado) * 100
porcentaje_no_pasaron

ESTADO
                                        NaN
AGUASCALIENTES                     2.439024
BAJA CALIFORNIA                    2.054795
BAJA CALIFORNIA SUR                     NaN
CAMPECHE                                NaN
CHIAPAS                            0.793651
CHIHUAHUA                          4.278075
CIUDAD DE MEXICO                   1.941748
COAHUILA DE ZARAGOZA                    NaN
COLIMA                                  NaN
DURANGO                            4.761905
GUANAJUATO                         5.000000
GUERRERO                           2.631579
HIDALGO                                 NaN
JALISCO                            0.864553
MEXICO                             1.506024
MICHOACAN DE OCAMPO                1.621622
MORELOS                            4.109589
NAYARIT                                 NaN
NUEVO LEON                         3.431373
OAXACA                             1.600000
PUEBLA                             3.960396
QUERETARO                

In [198]:
# Obtener la lista de todos los estados
todos_los_estados = df_question_2['ESTADO'].unique()

# Asegurarse de que todas las series tengan los mismos estados
conteo_no_pasaron_por_estado = conteo_no_pasaron_por_estado.reindex(todos_los_estados, fill_value=0)
conteo_total_por_estado = conteo_total_por_estado.reindex(todos_los_estados, fill_value=0)
porcentaje_no_pasaron = (conteo_no_pasaron_por_estado / conteo_total_por_estado).fillna(0) * 100

# Crear el DataFrame
df_estado_resumen = pd.DataFrame({
    'ESTADO': conteo_no_pasaron_por_estado.index,
    'No Pasaron': conteo_no_pasaron_por_estado.values,
    'Total': conteo_total_por_estado.values,
    'Porcentaje No Pasaron': porcentaje_no_pasaron.values
})

# Ordenar el DataFrame por 'Porcentaje No Pasaron' de mayor a menor
df_estado_resumen = df_estado_resumen.sort_values(by='Porcentaje No Pasaron', ascending=False).reset_index(drop=True)

# Mostrar el DataFrame resultante
print(df_estado_resumen)

                             ESTADO  No Pasaron  Total  Porcentaje No Pasaron
0                        GUANAJUATO          11    220               5.000000
1                           DURANGO           5    105               4.761905
2                          TLAXCALA           2     43               4.651163
3                         CHIHUAHUA           8    187               4.278075
4                           MORELOS           3     73               4.109589
5                            PUEBLA           4    101               3.960396
6   VERACRUZ DE IGNACIO DE LA LLAVE           4    116               3.448276
7                        NUEVO LEON           7    204               3.431373
8                          GUERRERO           1     38               2.631579
9                    AGUASCALIENTES           2     82               2.439024
10                       TAMAULIPAS           5    215               2.325581
11                  BAJA CALIFORNIA           3    146          

# Question 3
## ¿Cuáles son los estados que tienen más seguridad en sus despachadores y cuales son los que cuentan con la peor seguridad en sus despachadores?

**Fuga**
(Goteo constante de combustible)
Se inmoviliza el instrumento de medición cuando se observa escape de combustible en algún componente del sistema de despacho.

**Deficiencias evidentes**
(El instrumento presenta piezas sueltas o  desperfectos) 
El Dispensario debe funcionar sin presentar desperfectos, piezas sueltas u otras deficiencias, de tal forma que se encuentre apto para su uso.

In [199]:
test_columns_3 = [
    "CAUSAS DE INMOVILIZACION Seguridad 9.1",
    "CAUSAS DE INMOVILIZACION Seguridad 9.2",
    "ESTADO"
]
df_question_3 = df.copy()
nan_counts_3 = {}

for column in test_columns_3:
    if column == "ESTADO":
        df_question_3["ESTADO"] = df_question_3["ESTADO"].apply(lambda x: unidecode.unidecode(x).upper())
    else:
        df_question_3[column] = pd.to_numeric(df_question_3[column], errors='coerce')  # Coerce errors will set problematic entries to NaN
        nan_counts_3[column] = df_question_3[column].isna().sum()

print(nan_counts_3)    

{'CAUSAS DE INMOVILIZACION Seguridad 9.1': 2, 'CAUSAS DE INMOVILIZACION Seguridad 9.2': 2}


In [200]:
df_question_3 = df_question_3[test_columns_3]
df_question_3 = df_question_3[:-2]
df_question_3.tail(5)

CAUSAS DE INMOVILIZACION Seguridad 9.1  \
4114                                     0.0   
4115                                     0.0   
4116                                     0.0   
4117                                     0.0   
4118                                    11.0   

      CAUSAS DE INMOVILIZACION Seguridad 9.2               ESTADO  
4114                                     0.0               MEXICO  
4115                                     0.0               MEXICO  
4116                                     0.0               MEXICO  
4117                                     0.0  MICHOACAN DE OCAMPO  
4118                                    88.0

In [201]:
df_question_3 = df_question_3[df_question_3["ESTADO"] != ""]
conteo_toal_por_estado = df_question_3.groupby("ESTADO").size()
conteo_toal_por_estado

ESTADO
AGUASCALIENTES                      82
BAJA CALIFORNIA                    146
BAJA CALIFORNIA SUR                 37
CAMPECHE                            43
CHIAPAS                            126
CHIHUAHUA                          187
CIUDAD DE MEXICO                   103
COAHUILA DE ZARAGOZA                45
COLIMA                              43
DURANGO                            105
GUANAJUATO                         220
GUERRERO                            38
HIDALGO                            103
JALISCO                            347
MEXICO                             332
MICHOACAN DE OCAMPO                185
MORELOS                             73
NAYARIT                             61
NUEVO LEON                         204
OAXACA                             125
PUEBLA                             101
QUERETARO                          169
QUINTANA ROO                        39
SAN LUIS POTOSI                    120
SINALOA                            232
SONORA            

In [202]:
df_gasolineras_no_seguras = df_question_3[(df_question_3["CAUSAS DE INMOVILIZACION Seguridad 9.1"] != 0) | (df_question_3["CAUSAS DE INMOVILIZACION Seguridad 9.2"] != 0)]
df_gasolineras_no_seguras.head(10)

CAUSAS DE INMOVILIZACION Seguridad 9.1  \
236                                     0.0   
277                                     0.0   
304                                     0.0   
471                                     0.0   
482                                     0.0   
518                                     0.0   
527                                     0.0   
647                                     0.0   
664                                     0.0   
672                                     0.0   

     CAUSAS DE INMOVILIZACION Seguridad 9.2           ESTADO  
236                                     1.0          HIDALGO  
277                                     1.0           PUEBLA  
304                                     1.0           PUEBLA  
471                                     2.0       GUANAJUATO  
482                                     1.0          JALISCO  
518                                     1.0  SAN LUIS POTOSI  
527                                     2.0           MEXICO  
647                                     1.0          HIDALGO  
664                                     1.0       NUEVO LEON  
672                                     1.0           PUEBLA

In [203]:
df_gasolineras_no_seguras_por_estado = df_gasolineras_no_seguras.groupby("ESTADO").sum()
df_gasolineras_no_seguras_por_estado["Total fallas"] = df_gasolineras_no_seguras_por_estado[["CAUSAS DE INMOVILIZACION Seguridad 9.1", "CAUSAS DE INMOVILIZACION Seguridad 9.2"]].sum(axis=1)
df_gasolineras_no_seguras_por_estado.head(10)

CAUSAS DE INMOVILIZACION Seguridad 9.1  \
ESTADO                                                         
AGUASCALIENTES                                           0.0   
BAJA CALIFORNIA                                          2.0   
CHIAPAS                                                  0.0   
CHIHUAHUA                                                0.0   
COAHUILA DE ZARAGOZA                                     0.0   
DURANGO                                                  0.0   
GUANAJUATO                                               2.0   
HIDALGO                                                  2.0   
JALISCO                                                  0.0   
MEXICO                                                   1.0   

                      CAUSAS DE INMOVILIZACION Seguridad 9.2  Total fallas  
ESTADO                                                                      
AGUASCALIENTES                                           3.0           3.0  
BAJA CALIFORNIA                                          6.0           8.0  
CHIAPAS                                                  2.0           2.0  
CHIHUAHUA                                                9.0           9.0  
COAHUILA DE ZARAGOZA                                     1.0           1.0  
DURANGO                                                  2.0           2.0  
GUANAJUATO                                              12.0          14.0  
HIDALGO                                                  2.0           4.0  
JALISCO                                                  4.0           4.0  
MEXICO                                                   8.0           9.0

In [204]:
porcentaje_gasolineras_con_falla = (df_gasolineras_no_seguras_por_estado["Total fallas"] / conteo_total_por_estado) * 100

porcentaje_gasolineras_con_falla_ordenado = porcentaje_gasolineras_con_falla.sort_values(ascending=False)
df_porcentaje_gasolineras_con_falla = porcentaje_gasolineras_con_falla_ordenado.to_frame()
df_porcentaje_gasolineras_con_falla = df_porcentaje_gasolineras_con_falla.rename(columns={0: 'Porcentaje de Fallas'})
df_porcentaje_gasolineras_con_falla = df_porcentaje_gasolineras_con_falla.rename_axis('ESTADO').reset_index()

# Mostrar el DataFrame resultante
print(df_porcentaje_gasolineras_con_falla)


                             ESTADO  Porcentaje de Fallas
0                          TLAXCALA              6.976744
1                        GUANAJUATO              6.363636
2                   BAJA CALIFORNIA              5.479452
3                         CHIHUAHUA              4.812834
4                            OAXACA              4.000000
5                            PUEBLA              3.960396
6                           HIDALGO              3.883495
7                    AGUASCALIENTES              3.658537
8                         ZACATECAS              3.597122
9   VERACRUZ DE IGNACIO DE LA LLAVE              3.448276
10                          SINALOA              3.017241
11                        QUERETARO              2.958580
12                          MORELOS              2.739726
13                           MEXICO              2.710843
14             COAHUILA DE ZARAGOZA              2.222222
15                          DURANGO              1.904762
16            

## Resumen: 
### Los estados con mayor seguridad en sus despachadores de gasolina son:
- 24              BAJA CALIFORNIA SUR                   
- 25                         CAMPECHE                   
- 26                 CIUDAD DE MEXICO                   
- 27                           COLIMA                   
- 28                         GUERRERO                  
- 29                          NAYARIT                   
- 30                     QUINTANA ROO                   
- 31                          TABASCO                   
- 32                          YUCATAN

### Estados con peor seguridad en sus despachadores de gasolina:
- 0                      TLAXCALA                  6.97%
- 1                        GUANAJUATO              6.36%
- 2                   BAJA CALIFORNIA              5.47%
- 3                         CHIHUAHUA              4.81%
- 4                            OAXACA              4.00%

# Question 4
## ¿Las gasolineras con Golpe de presión en sus despachadores?

#### Golpe de presión
#### (La pantalla del dispensario censa combustible sin que el mismo se entregue)
#### Los golpes de presión generados por el flujo de combustible al activar la bomba, cuando aún permanece cerrada la válvula de descarga o pistola del sistema de despacho o dispensario, no deben afectar el registro de la pantalla que indica el combustible despachado.

In [205]:
df_question_4 = df.copy()
test_columns_4 = [
    'CAUSAS DE INMOVILIZACION Fallas Hidráulicas (5) 5.2',
    'ESTADO',
    'Dirección',
    'Colonia'
]

nan_counts_4 = {}

for c in test_columns_4:
    if c == "ESTADO" or c == 'Dirección' or c == 'Colonia':
        df_question_4[c] = df_question_4[c].apply(lambda x: unidecode.unidecode(x).upper())
    else:
        df_question_4[c] = pd.to_numeric(df_question_4[c], errors='coerce')  # Coerce errors will set problematic entries to NaN
        nan_counts_4[c] = df_question_4[c].isna().sum()

for key, value in nan_counts_4.items():
    print(key, value)

CAUSAS DE INMOVILIZACION Fallas Hidráulicas (5) 5.2 2


In [210]:
df_question_4.head(3)

CAUSAS DE INMOVILIZACION Fallas Hidráulicas (5) 5.2  ESTADO  \
0                                                0.0    MEXICO   
1                                                0.0    MEXICO   
2                                                0.0    MEXICO   

                                     Dirección              Colonia  
0                        AVENIDA MEXICO NO. 66          OLIMPICA 68  
1  AV. JARDINES DE MORELOS MZ 633 LT. 8-9 Y 10  JARDINES DE MORELOS  
2                   AV. CENTRAL LT. 1 MZ-A S/N         LAS AMERICAS

In [207]:
df_question_4 = df_question_4[test_columns_4]
df_question_4 = df_question_4[:-3]
df_question_4.tail(5)

CAUSAS DE INMOVILIZACION Fallas Hidráulicas (5) 5.2  \
4113                                                0.0     
4114                                                0.0     
4115                                                0.0     
4116                                                0.0     
4117                                                0.0     

                   ESTADO                                  Dirección  \
4113               MEXICO                GAS SEQUOIA, S. A. DE C. V.   
4114               MEXICO          PARADOR SAN JORGE, S. A. DE C. V.   
4115               MEXICO  FLORES ARCINIEGA NATALIA MARIA DEL CARMEN   
4116               MEXICO               JAIME IGNACIO VALDES GARDUNO   
4117  MICHOACAN DE OCAMPO               GRUPO OCTANO, S. A. DE C. V.   

                         Colonia  
4113         SAN BARTOLO MORELOS  
4114  ANTIGUO RANCHO SAN JOAQUIN  
4115         SAN ANDRES TIMILPAN  
4116                      CENTRO  
4117                 SIN COLONIA

In [208]:
df_gasolineras_golpe_de_presion = df_question_4[df_question_4['CAUSAS DE INMOVILIZACION Fallas Hidráulicas (5) 5.2'] != 0]
df_gasolineras_golpe_de_presion.info()

<class 'pandas.core.frame.DataFrame'>
Index: 3 entries, 397 to 2264
Data columns (total 4 columns):
 #   Column                                               Non-Null Count  Dtype  
---  ------                                               --------------  -----  
 0   CAUSAS DE INMOVILIZACION Fallas Hidráulicas (5) 5.2  3 non-null      float64
 1   ESTADO                                               3 non-null      object 
 2   Dirección                                            3 non-null      object 
 3   Colonia                                              3 non-null      object 
dtypes: float64(1), object(3)
memory usage: 120.0+ bytes


In [209]:
df_gasolineras_golpe_de_presion

CAUSAS DE INMOVILIZACION Fallas Hidráulicas (5) 5.2      ESTADO  \
397                                                 1.0    GUANAJUATO   
1655                                                2.0    NUEVO LEON   
2264                                                1.0       JALISCO   

                                              Dirección      Colonia  
397   CARRETERA QUERETARO - LEON KM 66 + 350 FRENTE ...          S/C  
1655                      CARRETERA A LAREDO KM. 20 SIN  CD. APODACA  
2264              SERVICIO PONCE-MEDINA, S. A. DE C. V.       CENTRO